tests/eve_tests/unit_tests/test_datamodels.py

```diff
@@ -156,21 +156,6 @@ def test_devtools_compatibility(example_model_factory):
         assert f"{name}=" in formatted_string
 
 
-def test_dataclass_compatibility(example_model_factory):
-    model = example_model_factory()
-    model_class = model.__class__
-
-    assert hasattr(model_class, "__dataclass_fields__") and isinstance(
-        model_class.__dataclass_fields__, dict
-    )
-    assert set(model_class.__datamodel_fields__.keys()) == set(
-        model_class.__dataclass_fields__.keys()
-    )
-    assert dataclasses.is_dataclass(model_class)
-    field_names = set(model_class.__datamodel_fields__.keys())
-    assert all(f.name in field_names for f in dataclasses.fields(model))
```

Comment:
Because the machinery to make a datamodel look like a dataclass has been also dropped, so there is no need to test it. I decided to drop the dataclass emulation because there is no real need for it and it's hard to maintain because the interface of dataclasses change in every python release, and when multiple python versions are supported, it is just a nightmare.